nityam_kumar/PostgresAssignment/src/controller/subject.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import uuid from "uniqid";
+import { Context } from "vm";
+import * as subjectService from "../services/subject";
+import * as teacherService from "../services/teacher";
+
+import { subjectIDSchema } from "../db/validateSchema/helperSchema";
+
+import AppError from "../utils/appError";
+
+import subjectSchema from "../db/validateSchema/subjectSchema";
+
+interface ISubject {
+  subject_id?: string;
+  subject_name: string;
+  teacher_id: string;
+  class_number: number;
+}
+
+export const createSubject = async (ctx: Context) => {
+  const s1: ISubject = ctx.request.body;
+  try {
+    await subjectSchema.validateAsync(s1);
+  } catch (err) {
+    throw new AppError(err.message, 400);
+  }
+
+  await Promise.all([
```

Comment:
what are we doing if any function returns true ???

Reply:
teacherService.checkExists(s1.teacher_id),
subjectService.checkAlreadyExist(s1),
before adding subject to database in subject table it must check whether teacher is already registered in teacher table and
current subject not already exists in subject table 
if any functions returns false and rejected then proper error will be displayed otherwise new subject will be created

Reply:
code changed